src/components/textfield/textfield.tsx

```diff
@@ -0,0 +1,46 @@
+import React from 'react';
+import styled from 'styled-components';
+import { TextfieldLabel } from './../typography/typography';
+
+const StyledLabel = styled(TextfieldLabel)`
+  display: block;
+  margin-bottom: 4px;
+
+  &.danger {
```

Comment:
I suggest to name this class something with `error` to not cause any confusion. The property is called `error` and it makes everything easier if the class name reflects that.